queries/c_sharp/highlights.scm

```diff
@@ -46,11 +64,22 @@
   (identifier) @type)
 
 (namespace_declaration
-  name: [
-    (qualified_name
-      (identifier) @module)
-    (identifier) @module
-  ])
+  name: (identifier) @module)
+
+(file_scoped_namespace_declaration
+  name: (identifier) @module)
+
+(qualified_name
+  (identifier) @module
+  (#not-has-ancestor? @module method_declaration)
+  (#not-has-ancestor? @module record_declaration)
+  (#has-ancestor? @module namespace_declaration))
```

Comment:
I think this can be combined with the below one, like `(#has-ancestor? @module namespace_declaration file_scoped_namespace_declaration))`? Also would it make sense to add `using_directive` to this list? (And then also I think we would have to add 
```scheme
(using_directive
  (identifier) @module)
```
but towards the end: I tested adding it next to this query and it wasn't applied with high enough priority)

Reply:
> I think this can be combined with the below one, like (#has-ancestor? @module namespace_declaration file_scoped_namespace_declaration))

Wasn't sure if that'd work - nice!

As for using directives, again you cannot assume it's a namespace - it probably is oftentimes but other times it could be just types

Reply:
> As for using directives, again you cannot assume it's a namespace - it probably is oftentimes but other times it could be just types

ahhh ok sry i get it now :sweat_smile: